src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2387,7 +2387,7 @@ async def lock(self, job, id, options):
             raise CallError(f'Please move system dataset to another pool before locking {id}')
 
         async def detach(delegate):
-            await delegate.stop((await delegate.query(self.__attachments_path(ds), True)))
+            await delegate.stop(await delegate.query(self.__attachments_path(ds), True))
```

Comment:
This is an async method calling a sync method. The sync method isn't doing anything that could block the event loop, but still isn't "proper". Could we convert `__attachments_path` to an async method? Or instead of changing that method, just duplicate what it's doing locally in this method?

Reply:
There is nothing wrong in calling sync methods in async methods if they are not doing IO or CPU-intensive computation.